core/src/test/scala/org/apache/spark/SSLOptionsSuite.scala

```diff
@@ -140,6 +140,24 @@ class SSLOptionsSuite extends SparkFunSuite {
     assert(opts.enabledAlgorithms === Set("ABC", "DEF"))
   }
 
+  test("test ssl settings are set only when ssl is enabled") {
+    val conf = new SparkConf
+    val hadoopConf = new Configuration()
+    conf.set("spark.ssl.enabled", "true")
+    conf.set("spark.ssl.keyStorePassword", "password")
+    conf.set("spark.ssl.ui.enabled", "false")
+    conf.set("spark.ssl.ui.keyStorePassword", "12345")
+
+
+    val defaultOpts = SSLOptions.parse(conf, hadoopConf, "spark.ssl", defaults = None)
+    val opts = SSLOptions.parse(conf, hadoopConf, "spark.ssl.ui", defaults = Some(defaultOpts))
```

Comment:
Could you use more explicitly meaningful variable names instead of `defaultOpts` and `opts`?